src/Generating/ComposableGenerator.cpp

```diff
@@ -525,7 +525,9 @@ void cComposableGenerator::InitFinishGens(cIniFile & a_IniFile)
 		}
 		else if (NoCaseCompare(finisher, "SprinkleFoliage") == 0)
 		{
-			m_FinishGens.push_back(cFinishGenPtr(new cFinishGenSprinkleFoliage(Seed)));
+			int MaxCactusHeight 	= a_IniFile.GetValueI("Plants", "MaxCactusHeight", 3);
+			int MaxSugarcaneHeight 	= a_IniFile.GetValueI("Plants", "MaxSugarcaneHeight", 3);
+			m_FinishGens.push_back(cFinishGenPtr(new cFinishGenSprinkleFoliage(Seed, MaxCactusHeight, MaxSugarcaneHeight)));
```

Comment:
I would suggest to use the [`std::make_shared`](http://en.cppreference.com/w/cpp/memory/shared_ptr/make_shared) function. [Stack Overflow](https://stackoverflow.com/questions/20895648/difference-in-make-shared-and-normal-shared-ptr-in-c/20895705#20895705) has a good explanation of the benefits.